src/Streamly/Internal/LZ4.hs

```diff
@@ -290,7 +290,7 @@ decompressChunk ctx arr = do
               let dstEnd = dstBegin `plusPtr`
                     (fromIntegral :: CInt -> Int) decompLen1
                   decompArr = MArray.Array fptr dstEnd dstMax
-              Array.unsafeFreeze <$> MArray.shrinkToFit decompArr
+              return $ Array.unsafeFreeze decompArr
```

Comment:
Add a comment: "We cannot shrink the array here, because that would reallocate the array invalidating the cached dictionary.".

Reply:
BTW, we should be able to shrink it safely when there is no prev block dependency.